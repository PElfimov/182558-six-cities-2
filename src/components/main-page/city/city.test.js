import React from 'react';
import City from './city';
import Enzyme, {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({adapter: new Adapter()});

const offers = [
  {
    isPremium: true,
    cost: 120,
    name: `Beautiful & luxurious apartment at great location`,
    rating: 55,
    type: `Apartment`
  },
  {
    isPremium: false,
    cost: 132,
    name: `Wood and stone place`,
    rating: 80,
    type: `Apartment`
  },
  {
    isPremium: true,
    cost: 120,
    name: `Canal View Prinsengracht`,
    rating: 55,
    type: `Apartment`
  },
  {
    isPremium: true,
    cost: 80,
    name: `Nice, cozy, warm big bed apartment`,
    rating: 100,
    type: `Apartment`
  },
  {
    isPremium: true,
    cost: 120,
    name: `Wood and stone place`,
    rating: 88,
    type: `Private room`
  }
];

it(`SignIn correctly renders after relaunch`, () => {
  const tree = shallow(<City offers={offers} handleHover={jest.fn()} />);

  expect(toJSON(tree)).toMatchSnapshot();
});
