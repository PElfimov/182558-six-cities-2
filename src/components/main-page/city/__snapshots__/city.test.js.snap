// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn correctly renders after relaunch 1`] = `
<Fragment>
  <div
    className="cities"
  >
    <div
      className="cities__places-container container"
    >
      <section
        className="cities__places places"
      >
        <h2
          className="visually-hidden"
        >
          Places
        </h2>
        <b
          className="places__found"
        >
          5
           places to stay in 
        </b>
        <WithSortList />
        <div
          className="cities__places-list places__list tabs__content"
        >
          <WithActiveCard
            key="undefined"
            offer={
              Object {
                "cost": 120,
                "isPremium": true,
                "name": "Beautiful & luxurious apartment at great location",
                "rating": 55,
                "type": "Apartment",
              }
            }
            onClick={[Function]}
            onHover={[MockFunction]}
          />
          <WithActiveCard
            key="undefined"
            offer={
              Object {
                "cost": 132,
                "isPremium": false,
                "name": "Wood and stone place",
                "rating": 80,
                "type": "Apartment",
              }
            }
            onClick={[Function]}
            onHover={[MockFunction]}
          />
          <WithActiveCard
            key="undefined"
            offer={
              Object {
                "cost": 120,
                "isPremium": true,
                "name": "Canal View Prinsengracht",
                "rating": 55,
                "type": "Apartment",
              }
            }
            onClick={[Function]}
            onHover={[MockFunction]}
          />
          <WithActiveCard
            key="undefined"
            offer={
              Object {
                "cost": 80,
                "isPremium": true,
                "name": "Nice, cozy, warm big bed apartment",
                "rating": 100,
                "type": "Apartment",
              }
            }
            onClick={[Function]}
            onHover={[MockFunction]}
          />
          <WithActiveCard
            key="undefined"
            offer={
              Object {
                "cost": 120,
                "isPremium": true,
                "name": "Wood and stone place",
                "rating": 88,
                "type": "Private room",
              }
            }
            onClick={[Function]}
            onHover={[MockFunction]}
          />
        </div>
      </section>
      <div
        className="cities__right-section"
      >
        <section
          className="cities__map map"
        >
          <PointsMap
            offers={
              Array [
                Object {
                  "cost": 120,
                  "isPremium": true,
                  "name": "Beautiful & luxurious apartment at great location",
                  "rating": 55,
                  "type": "Apartment",
                },
                Object {
                  "cost": 132,
                  "isPremium": false,
                  "name": "Wood and stone place",
                  "rating": 80,
                  "type": "Apartment",
                },
                Object {
                  "cost": 120,
                  "isPremium": true,
                  "name": "Canal View Prinsengracht",
                  "rating": 55,
                  "type": "Apartment",
                },
                Object {
                  "cost": 80,
                  "isPremium": true,
                  "name": "Nice, cozy, warm big bed apartment",
                  "rating": 100,
                  "type": "Apartment",
                },
                Object {
                  "cost": 120,
                  "isPremium": true,
                  "name": "Wood and stone place",
                  "rating": 88,
                  "type": "Private room",
                },
              ]
            }
          />
        </section>
      </div>
    </div>
  </div>
</Fragment>
`;
